rdr_service/api/ancillary_study_cloud_tasks_api.py

```diff
@@ -202,14 +203,36 @@ class WithdrawnParticipantNotifierTaskApi(BaseAncillaryTaskApi):
     Cloud Task endpoint: Send a slack alert to rdr-nph-alerts if the payload sent by partner
     contains information about participants who have withdrawn from AOU program.
     """
+
+    def __init__(self):
+        super().__init__()
+        self.rex_dao = RexParticipantMappingDao()
+        self.participant_dao = ParticipantDao()
+
     def post(self):
         super().post()
-        withdrawn_pids = self.data.get("withdrawn_pids")
+        nph_pids = self.data.get("nph_pids")
+        aou_pids = self._convert_nph_pids_to_aou_pids(nph_pids=nph_pids)
+        withdrawn_pids: list[str] = self.participant_dao.get_withdrawn_participant_ids(
+            participant_ids=aou_pids
+        )
+
         log_msg = f"{len(withdrawn_pids)} withdrawn PIDs received in NPH intake payload."
         logging.info(log_msg)
-
-        ts = CLOCK.now().strftime('%Y-%m-%d %H:%M:%S')
-        msg = (f"NPH Intake API Payload received on {ts} contains {len(withdrawn_pids)} withdrawn participant IDs:"
-               f" {', '.join(withdrawn_pids)}")
-        create_nph_incident(slack=True, message=msg)
+        if withdrawn_pids:
+            ts = CLOCK.now().strftime('%Y-%m-%d %H:%M:%S')
+            msg = (
+                f"NPH Intake API Payload received on {ts} contains {len(withdrawn_pids)} withdrawn participant IDs:"
+                f" {', '.join(withdrawn_pids)}"
+            )
+            create_nph_incident(slack=True, message=msg)
```

Comment:
I like the idea of making the alert more granular, but utilizing timestamps for searching will be arduous to track down if an investigation needs to be done. We are sent and store a uuid  that is generated from our partners that would be better here. Should be an instance variable called `bundle_identifier`.